doc/export_native_api.md

```diff
@@ -189,10 +189,13 @@ void foo2(wasm_exec_env_t exec_env,
 
 The runtime builder should ensure not broking the memory sandbox when exporting the native function to WASM. 
 
+A ground rule:
+
+- Do the pointer address conversion in the native API if "$\*" is not used for the pointer in the function signature
+
 A few recommendations:
 
 - Never pass any structure/class object pointer to native (do data serialization instead)
-- Do the pointer address conversion in the native API if "$\*" is not used for the pointer in the function signature 
 - Never pass a function pointer to the native 
```

Comment:
maybe add, "except if you plan to call it from native as a regular C/C++ callback. In that case, you must use wasm_runtime_call_indirect.".

What do you think?

Reply:
i don't understand what you mean. what's a "regular" callback? is there a "non-regular" callback?

Reply:
Edited for better description!

Reply:
i think this doc is basically about what the author recommends. i feel it's a bit awkward to explain more about what's discouraged here.
anyway, i added some clarifications.